pkg/controller/build/build_controller.go

```diff
@@ -339,18 +339,18 @@ func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b
 				if err = r.client.Update(ctx, &buildRun); err != nil {
 					return err
 				}
-				ctxlog.Info(ctx, fmt.Sprintf("succesfully updated BuildRun %s", buildRun.Name), namespace, buildRun.Namespace, name, buildRun.Name)
+				ctxlog.Info(ctx, fmt.Sprintf("successfully updated BuildRun %s", buildRun.Name), namespace, buildRun.Namespace, name, buildRun.Name)
 			}
 		}
-	case "false":
+	case "", "false":
```

Comment:
Thanks, just to be very clear, the switch is not covering the scenario when `build.AnnotationBuildRunDeletion` annotation is not present. I think we should not cover this in this `switch/case`, mainly because there is no need to run the underlying logic when the annotation is not defined.

I think you can use this PR, to validate prior to the switch if the annotation key is defined, see a quick example https://play.golang.org/p/FisuJkEk7D5

Reply:
Hi @qu1queee, I think your proposal will mean the following behavior:

- Create a build b1 without the annotation
- Create a buildrun br1 which references b1
-> the buildrun has no owner, that's fine

- Create a build b2 with annotation build.build.dev/build-run-deletion=true
- Create a buildrun br2 which references b2
-> the buildrun br2 will be owned by the build b2
- Update the build b2 to delete the annotation build.build.dev/build-run-deletion
-> the buildrun br2 still has b2 as owner although b2 does not anymore express that buildrun deletion should happen

Based on that, handling `""` in the same way as `"false"` is in my opinion accurate.

Reply:
I see, sorry it was not clear from the PR that this was the rational. I think the current code was a strict one, based on https://github.com/shipwright-io/build/pull/409#pullrequestreview-513326815 . I think this is ideal, but you will need to provide an integration test case for this flow. Use https://github.com/shipwright-io/build/blob/master/test/integration/build_to_buildruns_test.go#L344-L372 as a reference, but the patch part should be a removal of the annotation.